### PR TITLE
Fixed `databricks_pipeline` incorrect generation of `cluster` blocks

### LINKED ISSUE #1401: [ISSUE] databricks_pipeline does not support updating the resource
### Affected Resource(s)
https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/pipeline

### Expected Details
The field `cluster.custom_tags.cluster_type` is not listed as a required field (implied to be therefore optional), yet creating a DLT pipeline without at least one `cluster` block containing the following configuration will allow the pipeline to be created but NOT subsequently updated.
```
cluster {
    custom_tags {
        cluster_type = 'default'
    }
}
```

Attempting to update a pipeline created without the above block will return the error:
`Error: cannot update pipeline: clusters must have exactly 1 default cluster`

It seems that updates to the pipeline are trying to reference this tag which could be non-existent, not the `label` field.

**EDIT:** It seems that this is not a documentation issue, requiring the custom tag still will not allow the resource to be updated in subsequent `terraform apply`

### PR DESCRIPTION
**Pull request by alexott**
This fixes #1401 to unblock the production workloads.

But it should be investigated why we generated empty structs when using reflection

### CODE REVIEW

File: pipelines/resource_pipeline.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ const DefaultTimeout = 20 * time.Minute
 // 1. Pipeline clusters include a `Label` field.
 // 2. Spark version is not required (and shouldn't be specified) for pipeline clusters.
 // 3. num_workers is optional, and there is no single-node support for pipelines clusters.
+// When adding new attributes, also modify the isEmptyClusterStruct function below
```
Comment (nfx) [inline pipelines/resource_pipeline.go:25]: have you tried `reflect.IsZero` way?

https://www.geeksforgeeks.org/reflect-iszero-function-in-golang-with-examples/

**Review by nfx — CHANGES_REQUESTED** [review]

File: pipelines/resource_pipeline.go (hunk 1 above)
Comment (alexott) [reply pipelines/resource_pipeline.go:25]: I looked to reflect, but read that it may not handle pointers, etc. correctly

Comment (alexott) [reply pipelines/resource_pipeline.go:25]: I’ll look Monday

Comment (nfx) [reply pipelines/resource_pipeline.go:25]: i think it's more of an issue in `slice_set`, btw. e.g. diff is not properly picked up

Comment (alexott) [reply pipelines/resource_pipeline.go:25]: agree, but this requires more time to investigation. I had a hope to have quick & dirty fix, and then fix correctly when get time

**Comment by alexott** [comment]
It didn't have `suppress_diff` before: https://github.com/databricks/terraform-provider-databricks/commit/8260d907c58b97c4fa2b719c8a8267cb8cd81f05#diff-95aaa2c785925088721c159cdd5b07357cdc3b1899bf4e4d4e236460acf1b737L33

**Comment by alexott** [comment]
Test is added

File: pipelines/acceptance/pipeline_test.go [hunk 2]
```diff
@@ -81,14 +86,99 @@ func TestAccPipelineResource_CreatePipeline(t *testing.T) {
...
-					exclude = ["com.databricks.exclude"]
+				continuous = false
+			}
+			` + dltNotebookResource,
+		},
+	})
+}
+
+func TestAccAwsPipelineResource_CreatePipeline(t *testing.T) {
+	if cloud, ok := os.LookupEnv("CLOUD_ENV"); !ok || cloud != "AWS" {
```
Comment (nfx) [inline pipelines/acceptance/pipeline_test.go:97]:
```suggestion
	if cloud, ok := os.LookupEnv("CLOUD_ENV"); !ok || cloud != "aws" {
```

File: pipelines/acceptance/pipeline_test.go [hunk 3]
```diff
@@ -81,14 +86,99 @@ func TestAccPipelineResource_CreatePipeline(t *testing.T) {
...
+					key2 = "value2"
+				}
+				library {
+					notebook {
+						path = databricks_notebook.this.path
+					}
 				}
 
+				cluster {
+					instance_pool_id = "{var.COMMON_INSTANCE_POOL_ID}"
```
Comment (nfx) [inline pipelines/acceptance/pipeline_test.go:120]: we don't have this instance pool defined anywhere...

File: pipelines/acceptance/pipeline_test.go [hunk 4]
```diff
@@ -81,14 +86,99 @@ func TestAccPipelineResource_CreatePipeline(t *testing.T) {
...
+					num_workers = 2
+					custom_tags = {
+						cluster_type = "default"
+					}
+					aws_attributes {
+						first_on_demand = 1
+					}
+				}
+				cluster {
+					instance_pool_id = "{var.COMMON_INSTANCE_POOL_ID}"
```
Comment (nfx) [inline pipelines/acceptance/pipeline_test.go:131]: here as well

File: pipelines/acceptance/pipeline_test.go [hunk 5]
```diff
@@ -81,14 +86,99 @@ func TestAccPipelineResource_CreatePipeline(t *testing.T) {
...
+					key2 = "value2"
+				}
+				library {
+					notebook {
+						path = databricks_notebook.this.path
+					}
+				}
+
+				cluster {
+					instance_pool_id = "{var.COMMON_INSTANCE_POOL_ID}"
```
Comment (nfx) [inline pipelines/acceptance/pipeline_test.go:161]: here as well

File: pipelines/acceptance/pipeline_test.go [hunk 6]
```diff
@@ -81,14 +86,99 @@ func TestAccPipelineResource_CreatePipeline(t *testing.T) {
...
+					num_workers = 3
+					custom_tags = {
+						cluster_type = "default"
+					}
+					aws_attributes {
+						first_on_demand = 2
+					}
+				}
+				cluster {
+					instance_pool_id = "{var.COMMON_INSTANCE_POOL_ID}"
```
Comment (nfx) [inline pipelines/acceptance/pipeline_test.go:172]: here as well

**Review by nfx — CHANGES_REQUESTED** [review]

**Review by nfx — APPROVED** [review]
image

**Pull request merged**